server/hwapi/endpoints/certification/certification.py

```diff
@@ -50,7 +62,14 @@ def check_certification(
     Endpoint for checking certification status (whether a system is certified, not seen
     or some of its components have been seen on other systems)
     """
+    data: CertifiedResponse | RelatedCertifiedSystemExistsResponse | None
     certified, data = is_certified(system_info, db)
     if certified:
         return data
-    return NotCertifiedResponse(status=CertificationStatus.NOT_SEEN)
+    try:
+        partially_certifed, data = is_partially_certified(system_info, db)
+    except ValueError as exc:
```

Comment:
Do both is_certified() and is_partially_certified() need to be under this try/except block?

Reply:
For now, is_certified doesn’t raise any ValueErrors, but is_partially_certified does; that’s why only this method is wrapped with a try-except block. Do you think we should change it?

Reply:
Only if you think it might be able to raise that in the future. It just wasn't immediately clear to me if it was something we should be concerned with